src/components/calendar/index.js

```diff
@@ -0,0 +1,232 @@
+module.exports = {
+    template: require('calendar.html'),
+    data: function() {
+        return {
+            showSpinner: false,
+            spinnerMessage: ""
+        }
+    },
+    props: ['context', 'messages', 'importFile','shareWith'],
+    created: function() {
+        this.displaySpinner();
+        this.startListener();
+    },
+    methods: {
+	startListener: function() {
+	    var that = this;
+	    var iframe = document.getElementById("editor");
+	    if (iframe == null) {
+    		setTimeout(that.startListener, 1000);
+	    	return;
+	    }
+        // Listen for response messages from the frames.
+        window.addEventListener('message', function (e) {
+            // Normally, you should verify that the origin of the message's sender
+            // was the origin and source you expected. This is easily done for the
+            // unsandboxed frame. The sandboxed frame, on the other hand is more
+            // difficult. Sandboxed iframes which lack the 'allow-same-origin'
+            // header have "null" rather than a valid origin. This means you still
+            // have to be careful about accepting data via the messaging API you
+            // create. Check that source, and validate those inputs!
+            if (e.origin === "null" && e.source === iframe.contentWindow) {
+                if(e.data.type=="save") {
```

Comment:
This is the bit I want to focus on, the app api and keeping it as general as possible.